src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -730,18 +672,28 @@ async Task<InProgressState> CollapseInProgressStateAsync(InProgressState initial
                                 }
                                 else
                                 {
-                                    var metadataReference = await compilationState.GetMetadataReferenceAsync(
```

Comment:
@ToddGrun this loop (if you look upwards to the containing for-loop) is what is setting up p2p references.  It stays the same for normal states, but is tweaked for the frozen case. In the frozen case we just use the compilation-without-generated-docs (which is why we can be fast and not async here).